Edge/Cloudflare Deployments fail due to `api/_content/cache.123.json` size

### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.18.2
- Nuxt Version:     3.8.1
- CLI Version:      3.9.1
- Nitro Version:    2.7.2
- Package Manager:  npm@8.13.2
- Builder:          -
- User Config:      runtimeConfig, build, app, nitro, modules, tailwindcss, css, schemaOrg, vite, typescript, alias, experimental, ogImage
- Runtime Modules:  @pinia/nuxt@0.4.11, @nuxtjs/tailwindcss@6.8.0, nuxt-headlessui@1.1.4, nuxt-icon@0.3.3, @nuxt/content@2.9.0
- Build Modules:    -
------------------------------

### Reproduction

-

### Describe the bug

Cloudflare pages and I assume other edge products based on cf workers and KV are limited to 25mb in single file size. The cache fill nuxt content generates may easily surpass that.
-> 
```
✘ [ERROR] Error: Pages only supports files up to 26.2 MB in size
api/_content/cache.1699317964624.json is 47.5 MB in size

``` 

### Additional context

_No response_

### Logs

```Shell
1:51:57.934 | Found _routes.json in output directory. Uploading.
01:51:57.954 | Validating asset output directory
01:51:59.198 | Deploying your site to Cloudflare's global network...
01:52:06.292 |  
01:52:06.424 | ✘ [ERROR] Error: Pages only supports files up to 26.2 MB in size
01:52:06.425 |  
01:52:06.425 | api/_content/cache.1699317964624.json is 47.5 MB in size
```

--- a/src/module.ts
+++ b/src/module.ts
@@ -239,7 +239,7 @@ export interface ModuleOptions {
 
     /**
      * Control content cach generation.
-     * 
+     *
      * This option might be removed in the next major version.
      */
     cacheContents?: boolean

--- a/src/module.ts
+++ b/src/module.ts
@@ -424,15 +424,22 @@ export default defineNuxtModule<ModuleOptions>({
           method: 'get',
           route: `${options.api.baseURL}/query`,
           handler: resolveRuntimeModule('./server/api/query')
-        },
-        {
+        }
+      )
+
+      if (options.experimental?.clientDB) {
+        nitroConfig.handlers.push({
           method: 'get',
           route: nuxt.options.dev
             ? `${options.api.baseURL}/cache.json`
             : `${options.api.baseURL}/cache.${buildIntegrity}.json`,
           handler: resolveRuntimeModule('./server/api/cache')
+        })
+
+        if (!nuxt.options.dev) {
+          nitroConfig.prerender.routes.unshift(`${options.api.baseURL}/cache.${buildIntegrity}.json`)
         }
-      )
+      }
 
       if (options.experimental?.search) {
         const route = nuxt.options.dev
@@ -451,10 +458,6 @@ export default defineNuxtModule<ModuleOptions>({
           // Use text/plain to avoid Nitro render an index.html
           headers: options.experimental.search.indexed ? { 'Content-Type': 'text/plain' } : { 'Content-Type': 'application/json' }
         }
-      }
-
-      if (!nuxt.options.dev) {
-        nitroConfig.prerender.routes.unshift(`${options.api.baseURL}/cache.${buildIntegrity}.json`)
       }
 
       // Register source storages

--- a/src/module.ts
+++ b/src/module.ts
@@ -236,6 +236,13 @@ export interface ModuleOptions {
     clientDB?: boolean
     stripQueryParameters?: boolean
     advanceQuery?: boolean,
+
+    /**
+     * Control content cach generation.
+     * 
+     * This option might be removed in the next major version.
+     */
+    cacheContents?: boolean
     /**
      * Search mode.
      *
@@ -353,6 +360,7 @@ export default defineNuxtModule<ModuleOptions>({
     respectPathCase: false,
     experimental: {
       clientDB: false,
+      cacheContents: true,
       stripQueryParameters: false,
       advanceQuery: false,
       search: undefined
@@ -424,22 +432,15 @@ export default defineNuxtModule<ModuleOptions>({
           method: 'get',
           route: `${options.api.baseURL}/query`,
           handler: resolveRuntimeModule('./server/api/query')
-        }
-      )
-
-      if (options.experimental?.clientDB) {
-        nitroConfig.handlers.push({
+        },
+        {
           method: 'get',
           route: nuxt.options.dev
             ? `${options.api.baseURL}/cache.json`
             : `${options.api.baseURL}/cache.${buildIntegrity}.json`,
           handler: resolveRuntimeModule('./server/api/cache')
-        })
-
-        if (!nuxt.options.dev) {
-          nitroConfig.prerender.routes.unshift(`${options.api.baseURL}/cache.${buildIntegrity}.json`)
         }
-      }
+      )
 
       if (options.experimental?.search) {
         const route = nuxt.options.dev
@@ -458,6 +459,10 @@ export default defineNuxtModule<ModuleOptions>({
           // Use text/plain to avoid Nitro render an index.html
           headers: options.experimental.search.indexed ? { 'Content-Type': 'text/plain' } : { 'Content-Type': 'application/json' }
         }
+      }
+
+      if (!nuxt.options.dev) {
+        nitroConfig.prerender.routes.unshift(`${options.api.baseURL}/cache.${buildIntegrity}.json`)
       }
 
       // Register source storages

--- a/src/runtime/server/api/cache.ts
+++ b/src/runtime/server/api/cache.ts
@@ -9,7 +9,7 @@ export default defineEventHandler(async (event) => {
   const { content } = useRuntimeConfig()
   const now = Date.now()
   // Fetch all content
-  const contents = await serverQueryContent(event).find()
+  let contents = await serverQueryContent(event).find()
 
   // Generate Index
   await getContentIndex(event)
@@ -20,7 +20,7 @@ export default defineEventHandler(async (event) => {
   return {
     generatedAt: now,
     generateTime: Date.now() - now,
-    contents,
+    contents: content.experimental.cacheContents ? contents : [] as any,
     navigation
   }
 })

--- a/src/runtime/server/api/cache.ts
+++ b/src/runtime/server/api/cache.ts
@@ -9,7 +9,7 @@ export default defineEventHandler(async (event) => {
   const { content } = useRuntimeConfig()
   const now = Date.now()
   // Fetch all content
-  let contents = await serverQueryContent(event).find()
+  const contents = await serverQueryContent(event).find()
 
   // Generate Index
   await getContentIndex(event)

